traits/observers/_observer_path.py

```diff
@@ -0,0 +1,95 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+
+class ObserverPath:
```

Comment:
I don't know why I only realized this now: But this object actually represents *many* "paths", not just one.
An alternative name welcome...

Reply:
`ObserverGraph`?  Or even `ObserverDGraph`?  Trees can't have cycles and I am not sure we can confidently rule that possibility out at the moment...